### PR TITLE
Add uuid to sql invitations

### PR DESCRIPTION
**Pull request by orangejenny**
##### SUMMARY
https://dimagi-dev.atlassian.net/browse/ICDS-1308

Marking don't merge until I test a bit more.

### CODE REVIEW

File: corehq/apps/users/models.py
Comment (mkangia) [inline corehq/apps/users/models.py]: can we use `invitation_id` instead of uuid to keep in line with other models like SQLLocation?
this way we can also just keep the variable name `invitation_id` in other places that are getting updated.

**Review by mkangia — CHANGES_REQUESTED** [review]
I am guessing not but do we need to worry about the links that would get invalidated when this gets released? Do we need a roll out period to work both ways for a while?

File: corehq/apps/users/models.py
Comment (orangejenny) [reply corehq/apps/users/models.py]: I find it confusing that there's both a `SQLLocation.id` and a `SQLLocation.location_id`. Is that really a best practice that we use elsewhere?

Comment (mkangia) [reply corehq/apps/users/models.py]: I think we follow different things but introducing a new one might not be a good thing.
in some other instances,`ModuleBase` has `unique_id`, `CommCareCaseSQL` has `case_id`.

Comment (orangejenny) [reply corehq/apps/users/models.py]: Okay, I'm inclined to go with `unique_id` then. Agreed introducing a new convention isn't good.

Comment (esoergel) [reply corehq/apps/users/models.py]: > I find it confusing that there's both a SQLLocation.id and a SQLLocation.location_id. Is that really a best practice that we use elsewhere?

I definitely wouldn't call it a best practice.  And yes, it is confusing.  I only ever refer to `location.id` as `location.pk` in order to help with the ambiguity.  My rule of thumb is that you should pretty much only ever use `location_id` unless you're doing something low-level and db-related that needs access to the primary key (for instance looking across foreign key relationships without needing to do a join).

My recommendation would be to set `primary_key=True` on `uuid`, which will allow Django to drop `id` entirely.  Any place which would reference it would then reference `uuid` instead.  Do you know whether any other models have a foreign key reference to this one?  That would complicate the migration (that's why I never did it for locations).

Comment (esoergel) [reply corehq/apps/users/models.py]: No strong opinions on `unique_id` vs `uuid` vs `invitation_id`, though I do agree that following convention would be best.  You may be in a good position to _establish_ such a convention, actually.

Looking at the lines matching `id.=.*index` in `models.py` files, all results are <noun>_id.  There don't seem to be many models with a field specified as `primary_key=True`, so arguably that's a different concern.

Actually, after writing that out, I do have opinions - I like `uuid` the most, as I'd expect `id` to be a sequential ID.  `unique_id` is also reasonable.  `<noun>_id` feels like it should be used for related models, not as a primary key field (typing `location.location_id` makes my eye twitch).

Comment (millerdev) [reply corehq/apps/users/models.py]: > My recommendation would be to set `primary_key=True` on `uuid`

If you do this, please change the name to `id`

**Comment by orangejenny** [comment]
Added a few commits to this PR and to the final cleanup PR in this migration, https://github.com/dimagi/commcare-hq/pull/26687. I'm removing the old auto-incrementing id field and using `uuid` as the primary key. Not naming it `id` because I agree with @esoergel that within HQ there's a strong convention that `id` is an auto-incrementing integer.

> I am guessing not but do we need to worry about the links that would get invalidated when this gets released? Do we need a roll out period to work both ways for a while?

Yes, there are existing invitation emails that reference either the old SQL id or the older couch id. There's logic at the top of `UserInvitationView.__call__` to mitigate this. The last PR in this migration removes that logic, so it can't be merged until 30 days after this one is merged (invitations expire after 30 days).

**Review by esoergel — APPROVED** [review]

**Comment by orangejenny** [comment]
@mkangia Can you verify the changes you requested are addressed?

File: corehq/apps/users/views/__init__.py
Comment (mkangia) [inline corehq/apps/users/views/__init__.py]: Why we already have `id` and `couch_id`?

**Review by mkangia — APPROVED** [review]
Sorry, i was just reading via emails and didn't notice that i had blocked this PR.

File: corehq/apps/users/views/__init__.py
Comment (orangejenny) [reply corehq/apps/users/views/__init__.py]: Right now there are two kinds of invitations in the wild: old ones where the URL references and old couch id and new ones where the URL references a sql id. This PR will add a third kind, with URLs that reference the new uuid. So this code has to handle all three. https://github.com/dimagi/commcare-hq/pull/26687 cleans up this code.

Comment (mkangia) [reply corehq/apps/users/views/__init__.py]: woah, okay thanks.

**Pull request merged**
